kernel/spinlock.c

```diff
@@ -16,6 +16,20 @@ initlock(struct spinlock *lk, char *name)
   lk->cpu = 0;
 }
 
+uint
```

Comment:
Both `lock_test_and_set` and `lock_release` are not SMP-safe. You should add comments.